test/cql-pytest/cassandra_tests/functions/cast_fcts_test.py

```diff
@@ -15,9 +15,9 @@ def testInvalidQueries(cql, test_keyspace):
     with create_table(cql, test_keyspace, "(a int primary key, b text, c double)") as table:
         assertInvalidMessage(cql, table, "cannot be cast to", "SELECT CAST(a AS boolean) FROM %s")
 
-# Marked cassandra_bug because of CASSANDRA-18647 (see comment below)
-@pytest.mark.xfail(reason="issue #14508")
-def testNumericCastsInSelectionClause(cql, test_keyspace, cassandra_bug):
+# cast to the same type should be optimised
+@pytest.mark.xfail(reason="a new issue should be opened")
+def testOptimiseNumericCast(cql, test_keyspace, cassandra_bug):
```

Comment:
This test file was translated from Cassandra's unit tests - I don't want to rename the tests or start splitting them. This entire test was marked "xfail" because of issue 14508. If this was a mistake, and the test still fails after 14508 passes, where does it fail now? If there's a completely different bug, you should open a new issue - not write a comment "a new issue should be opened".

If you want to write a new short test just for 14508 and nothing else, please don't put it in this file (which should contain only tests translated from Cassandra) - rather put it in test/cql-pytest/test_cast_data.py. Thanks.

Reply:
@nyh  I guess Cassandra does some query optimizations and ignores cast to the same type. So CAST(int_col as int) is converted to int_col during the optimization and Cassandra returns just int_col as the column name.

Yes we can check the argument type in cast function and make a proper title here but I believe it's much better to create a new issue for query optimization.

btw could you please check https://github.com/scylladb/scylladb/issues/13250#issuecomment-1527104615

Reply:
Yes, it's indeed possible to have a patch that solves 90% of a problem and create a new issue for the 10% left especially if looks not important and you think that solving the 100% in one time is too ugly or something. However, if you do that I'm still asking that you don't split or rename the tests in the `cassandra_tests` subdirectory, and instead create new tests specifically for the new issue, outside the `cassandra_tests` directory.

Reply:
I can suggest two options:
1. You can fix the 10% remaining right now - if the cast is to the same type, drop it when printing the function name.
2. If you leave that last problem (cast to the same type's name), you can leave the test in cassandra_tests unmodified (it will sstill fail one the same issue...) and instead create two new tests in test/cql-pytest/ test_cast_data.py - one test will test the name of the cast to a different type and will pass, and the second test will test the name of the cast to the same time, and will xfail (and pass on Cassandra). If you do this, please note that the coding style in test_cast_data.py is different from the translated Cassandra tests. In particular, please try to use an existing table (we have "table1" there) or create a new shared table instead of creating a new table for each test.

Reply:
@nyh fixed

Reply:
> I can suggest two options:
> 
> 1. You can fix the 10% remaining right now - if the cast is to the same type, drop it when printing the function name.

Hope I managed to optimize a query with cast to the same type. As result correct name is printed. Could you please check it?